src/task.c

```diff
@@ -659,32 +659,33 @@ jl_function_t *jl_unprotect_stack_func;
-                                   jl_any_type,
-                                   jl_emptysvec,
-                                   jl_svec(13,
-                                           jl_symbol("parent"),
-                                           jl_symbol("storage"),
-                                           jl_symbol("state"),
-                                           jl_symbol("consumers"),
-                                           jl_symbol("donenotify"),
-                                           jl_symbol("result"),
-                                           jl_symbol("exception"),
-                                           jl_symbol("backtrace"),
-                                           jl_symbol("code"),
-                                           jl_symbol("ctx"),
-                                           jl_symbol("bufsz"),
-                                           jl_symbol("stkbuf"),
-                                           jl_symbol("ssize")),
-                                   jl_svec(13,
-                                           jl_any_type,
-                                           jl_any_type, jl_sym_type,
-                                           jl_any_type, jl_any_type,
-                                           jl_any_type, jl_any_type,
-                                           jl_any_type, jl_any_type,
-                                           jl_tupletype_fill(sizeof(jl_jmp_buf), (jl_value_t*)jl_uint8_type),
-                                           jl_long_type, jl_voidpointer_type, jl_long_type),
-                                   0, 1, 8);
+    jl_task_type = (jl_datatype_t*)
+        jl_new_datatype(jl_symbol("Task"),
+                        jl_any_type,
+                        jl_emptysvec,
+                        jl_svec(13,
+                                jl_symbol("parent"),
+                                jl_symbol("storage"),
+                                jl_symbol("state"),
+                                jl_symbol("consumers"),
+                                jl_symbol("donenotify"),
+                                jl_symbol("result"),
+                                jl_symbol("exception"),
+                                jl_symbol("backtrace"),
+                                jl_symbol("code"),
+                                jl_symbol("ctx"),
+                                jl_symbol("bufsz"),
+                                jl_symbol("stkbuf"),
+                                jl_symbol("ssize")),
+                        jl_svec(13,
+                                jl_any_type,
+                                jl_any_type, jl_sym_type,
+                                jl_any_type, jl_any_type,
+                                jl_any_type, jl_any_type,
+                                jl_any_type, jl_any_type,
+                                jl_tupletype_fill(sizeof(jl_jmp_buf), (jl_value_t*)jl_uint8_type),
```

Comment:
It's not a new bug, but I just noticed that this tuple doesn't have the right alignment. Either the number of pointers before it should be divisible by 4, or its eltype should be whatever the platform expects, or we should hide these last 4 fields, in addition to all of the ones that come after them.